Scripts/Runtime/CallbackHandlers/WitResponseMatcher.cs

```diff
@@ -0,0 +1,278 @@
+                                {
+                                    result = "";
+                                    break;
+                                }
+                            }
+                        }
+
+                        if (!string.IsNullOrEmpty(result))
+                        {
+                            formatEvent.onFormattedValueEvent?.Invoke(result);
+                        }
+                    }
+                }
+
+                List<string> values = new List<string>();
+                for (int i = 0; i < references.Length; i++)
+                {
+                    var reference = references[i];
+                    var value = reference.GetStringValue(response);
+                    values.Add(value);
+                }
+
+                onMultiValueEvent.Invoke(values.ToArray());
+            }
+        }
+
+        private bool ValueMatches(WitResponseNode response)
+        {
+            bool matches = true;
+            for (int i = 0; i < valueMatchers.Length && matches; i++)
+            {
+                var matcher = valueMatchers[i];
+                var value = references[i].GetStringValue(response);
+                matches &= !matcher.contentRequired || !string.IsNullOrEmpty(value);
+
+                switch (matcher.matchMethod)
+                {
+                    case MatchMethod.RegularExpression:
+                        matches &= Regex.Match(value, matcher.matchValue).Success;
+                        break;
+                    case MatchMethod.Text:
+                        matches &= value == matcher.matchValue;
+                        break;
+                    case MatchMethod.IntegerComparison:
+                        matches &= CompareInt(value, matcher.matchValue, matcher.comparisonMethod);
+                        break;
+                    case MatchMethod.FloatComparison:
+                        matches &= CompareFloat(value, matcher.matchValue, matcher.comparisonMethod);
+                        break;
+                    case MatchMethod.DoubleComparison:
+                        matches &= CompareDouble(value, matcher.matchValue, matcher.comparisonMethod);
+                        break;
+                }
+            }
+
+            return matches;
+        }
+
+        private bool CompareDouble(string value, string matchValue,
+            ComparisonMethod comparisonMethod)
+        {
+            double dValue;
+
+            // This one is freeform based on the input so we will retrun false if it is not parsable
+            if (!double.TryParse(value, out dValue)) return false;
+
+            // We will throw an exception if match value is not a numeric value. This is a developer
+            // error.
+            double dMatchValue = double.Parse(matchValue);
+
+            switch (comparisonMethod)
+            {
+                case ComparisonMethod.Equals:
+                    return Math.Abs(dValue - dMatchValue) < .0001f;
```

Comment:
Const for this? Since it's used several times.

Reply:
Yeah I was debating adding a tolerance field. I think I will.